manim/animation/composition.py

```diff
@@ -106,34 +106,13 @@ def interpolate(self, alpha):
 
 
 class Succession(AnimationGroup):
+    """
+    Play several animations in a row. 
+    """
     CONFIG = {
         "lag_ratio": 1,
     }
 
-    def begin(self):
-        assert(len(self.animations) > 0)
-        self.init_run_time()
-        self.active_animation = self.animations[0]
-        self.active_animation.begin()
-
-    def finish(self):
-        self.active_animation.finish()
-
-    def update_mobjects(self, dt):
-        self.active_animation.update_mobjects(dt)
-
-    def interpolate(self, alpha):
-        index, subalpha = integer_interpolate(
-            0, len(self.animations), alpha
-        )
-        animation = self.animations[index]
-        if animation is not self.active_animation:
-            self.active_animation.finish()
-            animation.begin()
-            self.active_animation = animation
-        animation.interpolate(subalpha)
```

Comment:
I don't think it's safe to remove this logic since this version calls `finish()` on each of the animations after their runtime expires but by only setting `lag_ratio = 1` that wouldn't happen.

Reply:
You're right, it's not safe and I just noticed (thanks to the tests) that it has broken `ShowCreationThenFadeOut` (it inherits from `Succession`). In fact, I think that this logic is necessary when using two animations that `self.finish()` either create or destroy a object. But for a strange reason, it doesn't work for any other couple of animation. 
I mark this PR as a draft and will work on a fix again.  
thank you for the review!